frontend/src/metabase/query_builder/components/filters/pickers/DatePicker/DatePicker.unit.spec.tsx

```diff
@@ -0,0 +1,409 @@
+import React, { useState } from "react";
+import { render, screen, fireEvent } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+import { ORDERS } from "__support__/sample_database_fixture";
+import Filter from "metabase-lib/lib/queries/structured/Filter";
+
+import DatePicker from "./DatePicker";
+
+const ordersQuery = ORDERS.query();
+
+// this component does not manage its own filter state, so we need a wrapper to test
+// any state updates because the component's behavior is based on the filter state
+const DatePickerStateWrapper = ({
+  filter,
+  onCommit = jest.fn(),
+  onChange = jest.fn(),
+}: {
+  filter: Filter;
+  onCommit?: (arg: any) => void;
+  onChange?: (arg: any) => void;
+}) => {
+  const [filterValue, setFilterValue] = useState(filter);
+  return (
+    <DatePicker
+      filter={filterValue}
+      onFilterChange={(arg: any) => {
+        setFilterValue(arg);
+        onChange(arg);
+      }}
+      onCommit={onCommit}
+    />
+  );
+};
+
+describe("DatePicker", () => {
+  beforeAll(() => {
+    // this should keep these tests from behaving differently when run in the future
+    jest.useFakeTimers();
+    jest.setSystemTime(new Date("2020-05-01 08:00:00"));
+  });
+
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+
+  const dummyFunction = jest.fn();
+
+  describe("Static States", () => {
+    it("renders a date picker component", () => {
+      const filter = new Filter(
+        [null, ["field", ORDERS.CREATED_AT.id, null]],
+        null,
+        ordersQuery,
+      );
+
+      render(
+        <DatePicker
+          filter={filter}
+          onFilterChange={dummyFunction}
+          onCommit={dummyFunction}
+        />,
+      );
+
+      screen.getByTestId("date-picker");
```

Comment:
I think this should better be wrapped in `expect().toBeInTheDocument()` to get a more readable error message?